backend/src/routes/content.rs

```diff
@@ -0,0 +1,413 @@
+            .expect("Cannot save content");
+        // println!("{:?}", res1.post_id.unwrap());
+        // fill the row in content_reply
+        let content_reply = pgdb::content_reply::ActiveModel {
+            post_id: Set(res1.post_id.unwrap()),
+            reply_id: Set(0),
+            author: Set(content.author.to_string()),
+            anonymous: Set(content.anonymous),
+            created_time: Set(created_time.with_timezone(&FixedOffset::east(8 * 3600))),
+            modified_time: Set(modified_time.with_timezone(&FixedOffset::east(8 * 3600))),
+            content: Set(Some(content.content.to_string()).to_owned()),
+            ..Default::default()
+        };
+        // insert the row into database
+        let res2 = content_reply
+            .insert(&pg_con)
+            .await
+            .expect("Cannot save content");
+        // println!("{:?}", res2.reply_id.unwrap());
+        // return the response
+        post_response.success = true;
+        post_response.post_id = res2.post_id.unwrap();
+        (Status::Ok, Json(post_response))
+    }
+}
+
+#[get("/<post_id>/<page>")]
+pub async fn content_read(
+    db: Connection<PgDb>,
+    post_id: i32,
+    page: i32,
+) -> (Status, Json<ReadResponse>) {
+    let pg_con = db.into_inner();
+    // create a response struct
+    let mut read_response = ReadResponse {
+        success: false,
+        error: Vec::new(),
+        subject_info: Subject {
+            post_id: post_id,
+            title: String::new(),
+            author: String::new(),
+            anonymous: false,
+            created_time: String::new(),
+            modified_time: String::new(),
+            section: String::new(),
+            tag1: String::new(),
+            tag2: String::new(),
+            tag3: String::new(),
+            post_len: 0i32,
+        },
+        reply_info: Vec::new(),
+    };
+    // check if the post not exsits, add corresponding error if so
+    let subject_info = ContentSubject::find_by_id(post_id)
+        .one(&pg_con)
+        .await
+        .expect("cannot fetch content from pgdb");
+    if subject_info == None {
+        read_response.error.push("Post not exsits".to_string());
+        (Status::BadRequest, Json(read_response))
+    } else {
+        // get subject data
+        let subject_data = subject_info.unwrap();
+        // the number of replies displayed on one page
+        let item_display = 10;
+        let reply_info: Vec<pgdb::content_reply::Model>;
+        if subject_data.post_len < page * item_display {
+            reply_info = ContentReply::find()
+                .filter(
+                    Condition::all()
+                        .add(pgdb::content_reply::Column::PostId.eq(post_id))
+                        .add(pgdb::content_reply::Column::ReplyId.gte((page - 1) * item_display)),
+                )
+                .order_by_asc(pgdb::content_reply::Column::ReplyId)
+                .all(&pg_con)
+                .await
```

Comment:
You can use paginator:

https://www.sea-ql.org/SeaORM/docs/basic-crud/select#paginate-result

https://docs.rs/sea-orm/0.3.2/sea_orm/struct.Paginator.html